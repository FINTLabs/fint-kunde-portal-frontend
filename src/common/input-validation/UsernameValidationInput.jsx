import React from "react";
import { styled } from "@mui/material/styles";
import {
  FormControl,
  FormHelperText,
  Input,
  InputLabel,
    Grid,
    Typography
} from "@mui/material";
import PropTypes from "prop-types";

const PREFIX = 'UsernameValidationInput';

const classes = {
  realm: `${PREFIX}-realm`,
  label: `${PREFIX}-label`,
  input: `${PREFIX}-input`
};

const StyledFormControl = styled(FormControl)((
  {
    theme
  }
) => ({
  [`& .${classes.realm}`]: {
    width: "50%",
    display: "box",
    boxAlign: "center",
    fontSize: "1rem"
  },

  [`& .${classes.label}`]: {
    width: "50%",
    position: "inherit"
  },

  [`& .${classes.input}`]: {
    width: "50%"
  }
}));

class UsernameValidationInput extends React.Component {
  onChangeUsername = event => {
    let username = event.target.value;
    let usernameValidator = new RegExp("^[a-zA-Z0-9_-]{3,128}$");
    let valid = usernameValidator.test(username);

    this.setState({
      usernameValid: valid
    });

    this.props.usernameIsValid(valid);
    this.props.onChange(event);
  };

  constructor(props) {
    super(props);
    this.state = {
      usernameValid: true
    };
  }

  render() {
    const { name, title, realm, } = this.props;
    return (
      <StyledFormControl fullWidth error={!this.state.usernameValid} required>
        <InputLabel className={classes.label} htmlFor={name}>
          {title}
        </InputLabel>
        <Grid container>
          <Input
            autoFocus
            name={name}
            onChange={this.onChangeUsername}
            className={classes.input}
            id={"userNameInput"}
          />
          <Typography variant="caption" className={classes.realm}>
            {realm}
          </Typography>
        </Grid>

        <FormHelperText>
          {this.state.usernameValid
            ? ""
            : "Brukernavnet kan bare inneholde a-z, A-Z, 0-9, - og _. Det kan fra 3-128 tegn langt."}
        </FormHelperText>
      </StyledFormControl>
    );
  }
}

UsernameValidationInput.defaultProps = {
  realm: "@dummy.no"
};
UsernameValidationInput.propTypes = {
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  realm: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  usernameIsValid: PropTypes.func.isRequired
};
export default (UsernameValidationInput);
