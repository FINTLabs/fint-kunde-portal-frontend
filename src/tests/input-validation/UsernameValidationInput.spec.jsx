import React from "react";
import { create } from "react-test-renderer";
import UsernameValidationInput from "../../common/input-validation/UsernameValidationInput";
describe("Feature component", () => {
  test("it matches the snapshot", () => {
    const component = create(
      <UsernameValidationInput
        name="test"
        className="test"
        title="test"
        usernameIsValid={()=>{}}
        onChange={()=>{}}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
