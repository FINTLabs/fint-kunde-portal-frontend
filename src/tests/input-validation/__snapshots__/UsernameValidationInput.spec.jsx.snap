// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature component it matches the snapshot 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root UsernameValidationInput-label-2 MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
    data-shrink={false}
    htmlFor="test"
  >
    test
    <span
      className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
    >
      â€‰
      *
    </span>
  </label>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline UsernameValidationInput-input-3 MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={true}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="test"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
    <span
      className="MuiTypography-root UsernameValidationInput-realm-1 MuiTypography-caption"
    >
      @dummy.no
    </span>
  </div>
  <p
    className="MuiFormHelperText-root Mui-required"
  >
    
  </p>
</div>
`;
